plan/gen_physical_plans.go

```diff
@@ -140,7 +140,52 @@ func (p *LogicalJoin) getMergeJoin(prop *requiredProp) []PhysicalPlan {
 			joins = append(joins, mergeJoin)
 		}
 	}
-	return joins
+	// If there are some joinKeys in children property.
+	if len(joins) > 0 {
+		return joins
+	}
+	// Otherwise, generate enforced merge join plan.
+	return p.getEnforcedMergeJoin(prop)
```

Comment:
we only need to call this function when `(p.preferJoinType & preferMergeJoin) > 0` is true